### ISSUE TITLE
clang -Wcomma warning in public header cvdef.h

### ISSUE THREAD

**Issue by seanm** [opencv/opencv#12933]
##### System information (version)
- OpenCV => git master
- Operating System / Platform => macOS
- Compiler => clang

##### Detailed description

current OpenCV git master code produces a -Wcomma warning with clang:

/path/to/include/opencv4/opencv2/core/cvdef.h:731:53: warning: possible misuse of comma operator here [-Wcomma]

It seems to have been introduced by 80b62a41c6 by Vadim Pisarevsky on 2018-09-06.

False positive or not, it would be nice to reformulate this to not cause the warning, because:
- this is a public header, and thus affect projects that use Wcomma and build cleanly otherwise.
- it is the only -Wcomma warning.
- 3.4.3's public headers have no -Wcomma warnings.

**Comment by alalek** [comment]
There is common practice in GCC/Clang world to include external headers paths via "-isystem" flag.
Exactly to ignore any "external warnings".

**Comment by seanm** [comment]
@alalek, yes, that's certainly a possibility, though not always desirable.

I forgot to add that there's an easy code change to suppress the warning, as indicated by clang's full output, which I should have included:

```
/path/to/include/opencv4/opencv2/core/cvdef.h:731:53: warning: possible misuse of comma operator here [-Wcomma]
                 e == 0 ? (out.f -= 6.103515625e-05f, out.u) : t) | sign;
                                                    ^
/path/to/include/opencv4/opencv2/core/cvdef.h:731:28: note: cast expression to void to silence warning
                 e == 0 ? (out.f -= 6.103515625e-05f, out.u) : t) | sign;
                           ^~~~~~~~~~~~~~~~~~~~~~~~~
                           static_cast<void>(       )
```

I've confirmed adding the static_cast<void> works, and can create an MR if you'll accept it.

**Comment by alalek** [comment]
Feel free to create PR.

**Comment by seanm** [comment]
I just pulled newest before working on a patch, and it seems all that code is now gone entirely...
